File: taskmanager/src/test/java/com/example/alexeyglushkov/taskmanager/task/PriorityTaskProviderWithRestorableProviderTest.java
package com.example.alexeyglushkov.taskmanager.task;

import android.os.Handler;
import android.os.Looper;

/**
 * Created by alexeyglushkov on 18.06.17.
 */

public class PriorityTaskProviderWithRestorableProviderTest extends PriorityTaskProviderTest{
    @Override
    protected TaskProvider prepareTaskProvider() {
        return new RestorableTaskProvider(new PriorityTaskProvider(new Handler(Looper.myLooper()), "TestId"));
    }

    protected PriorityTaskProvider getPriorityTaskProvider() {
        RestorableTaskProvider restorableTaskProvider = (RestorableTaskProvider)taskProvider;
        return (PriorityTaskProvider)restorableTaskProvider.getProvider();
    }
}

File: taskmanager/src/test/java/com/example/alexeyglushkov/taskmanager/task/PriorityTaskProviderWithRestorableProviderTest.kt
package com.example.alexeyglushkov.taskmanager.task

import android.os.Handler
import android.os.Looper

/**
 * Created by alexeyglushkov on 18.06.17.
 */

class PriorityTaskProviderWithRestorableProviderTest : PriorityTaskProviderTest() {

    override val priorityTaskProvider: PriorityTaskProvider
        get() {
            val restorableTaskProvider = taskProvider as RestorableTaskProvider
            return restorableTaskProvider.provider as PriorityTaskProvider
        }

    override fun prepareTaskProvider(): TaskProvider {
        return RestorableTaskProvider(PriorityTaskProvider(Handler(Looper.myLooper()), "TestId"))
    }
}
